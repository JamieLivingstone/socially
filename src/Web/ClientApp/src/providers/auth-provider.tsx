import React, { ReactNode } from 'react';

import { Loading } from '../components';
import { AuthContext, useProvideAuth } from '../hooks';

export function AuthProvider({ children }: { children: ReactNode }) {
  const auth = useProvideAuth();

  if (auth.initializing) {
    return <Loading />;
  }

  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>;
}
