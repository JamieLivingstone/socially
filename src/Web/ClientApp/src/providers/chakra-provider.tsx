import { ChakraProvider as ChakraProviderBase, extendTheme, withDefaultColorScheme } from '@chakra-ui/react';
import React, { ReactNode } from 'react';

const theme = extendTheme(withDefaultColorScheme({ colorScheme: 'green' }), {
  initialColorMode: 'light',
  useSystemColorMode: false,
});

export function ChakraProvider({ children }: { children: ReactNode }) {
  return <ChakraProviderBase theme={theme}>{children}</ChakraProviderBase>;
}
