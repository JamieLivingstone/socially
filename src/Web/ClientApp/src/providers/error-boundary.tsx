import axios from 'axios';
import React, { Component } from 'react';

import { Error } from '../screens/miscellaneous/error';
import { NotFound } from '../screens/miscellaneous/not-found';

type ErrorBoundaryProps = {
  children: React.ReactNode;
};

type ErrorBoundaryState = {
  hasError: boolean;
  error: Error | null;
};

export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false, error: null };
  }

  static getDerivedStateFromError(error: Error) {
    return { hasError: true, error };
  }

  render() {
    if (this.state.hasError) {
      if (axios.isAxiosError(this.state.error) && this.state.error.response?.status === 404) {
        return <NotFound />;
      }

      return <Error />;
    }

    return this.props.children;
  }
}
