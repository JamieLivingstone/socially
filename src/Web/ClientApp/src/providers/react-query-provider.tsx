import axios from 'axios';
import React, { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

const RETRY_STATUS_CODES = [408, 413, 429, 500, 502, 503, 504];

const client = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
      retry: (failureCount, error) => {
        if (
          axios.isAxiosError(error) &&
          error.response &&
          failureCount < 3 &&
          RETRY_STATUS_CODES.includes(error.response.status)
        ) {
          return true;
        }

        return false;
      },
    },
  },
});

export function ReactQueryProvider({ children }: { children: ReactNode }) {
  return <QueryClientProvider client={client}>{children}</QueryClientProvider>;
}
