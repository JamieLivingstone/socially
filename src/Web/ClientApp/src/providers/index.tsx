import React, { ReactNode } from 'react';
import { BrowserRouter } from 'react-router-dom';

import { AuthProvider } from './auth-provider';
import { ChakraProvider } from './chakra-provider';
import { ErrorBoundary } from './error-boundary';
import { ReactQueryProvider } from './react-query-provider';

export function Providers({ children }: { children: ReactNode }) {
  return (
    <ReactQueryProvider>
      <ChakraProvider>
        <ErrorBoundary>
          <AuthProvider>
            <BrowserRouter>{children}</BrowserRouter>
          </AuthProvider>
        </ErrorBoundary>
      </ChakraProvider>
    </ReactQueryProvider>
  );
}
