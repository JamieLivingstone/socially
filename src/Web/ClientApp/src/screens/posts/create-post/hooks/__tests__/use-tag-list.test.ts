import { act, renderHook } from '@testing-library/react-hooks';
import moxios from 'moxios';
import { expect } from 'vitest';

import { TagListResponse, useTagList } from '../';
import { Providers } from '../../../../../providers';

describe('useTagList', () => {
  test('returns paginated tag list when a filter is specified', async () => {
    const response: TagListResponse = {
      hasNextPage: true,
      hasPreviousPage: false,
      items: [{ name: 'tag-one' }],
      pageNumber: 1,
      totalCount: 3,
      totalPages: 3,
    };

    moxios.stubRequest(new RegExp('/api/v1/tags.*'), {
      status: 200,
      response,
    });

    const { result, waitFor } = renderHook(() => useTagList(), { wrapper: Providers });

    expect(result.current.tags.length).toEqual(0);

    act(() => result.current.setFilter('tag-o'));

    await waitFor(() => result.current.tags.length > 0);

    expect(result.current).toMatchSnapshot();
  });
});
