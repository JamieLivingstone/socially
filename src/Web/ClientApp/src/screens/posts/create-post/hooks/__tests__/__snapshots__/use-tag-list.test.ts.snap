// Vitest Snapshot v1

exports[`useTagList > returns paginated tag list when a filter is specified 1`] = `
{
  "error": null,
  "isError": false,
  "isLoading": false,
  "refetch": [Function],
  "setFilter": [Function],
  "tags": [
    {
      "name": "tag-one",
    },
  ],
}
`;
